sentry-android-core/src/main/java/io/sentry/android/core/SentryAndroidDateProvider.java

```diff
@@ -0,0 +1,30 @@
+package io.sentry.android.core;
+
+import android.os.Build;
+import io.sentry.SentryDate;
+import io.sentry.SentryDateProvider;
+import io.sentry.SentryInstantDateProvider;
+import io.sentry.SentryNanotimeDateProvider;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * This {@link SentryDateProvider} will use {@link java.time.Instant} for Android API 26+ or a
+ * combination of {@link java.util.Date} and System.nanoTime() on lower versions.
+ */
+public final class SentryAndroidDateProvider implements SentryDateProvider {
+
+  private final @NotNull SentryDateProvider dateProvider;
+
+  public SentryAndroidDateProvider() {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
H: Use `BuildInfoProvider` to make it testable.

Reply:
Also added a test, now that it's easy to test this.